packages/vsx-registry/src/browser/vsx-extensions-model.ts

```diff
@@ -232,30 +232,52 @@ export class VSXExtensionsModel {
                 if (!allVersions) {
                     continue;
                 }
-                const res = await client.query({ extensionId: id, extensionVersion: allVersions.version, includeAllVersions: true });
-                let verified = res.extensions?.[0].verified;
-                if (!verified) {
-                    if (res.extensions?.[0].publishedBy.loginName === 'open-vsx') {
-                        verified = true;
-                    }
-                }
-                if (!this.preferences.get('extensions.onlyShowVerifiedExtensions') || verified) {
-                    this.setExtension(id).update(Object.assign(data, {
-                        publisher: data.namespace,
-                        downloadUrl: data.files.download,
-                        iconUrl: data.files.icon,
-                        readmeUrl: data.files.readme,
-                        licenseUrl: data.files.license,
-                        version: allVersions.version,
-                        verified: verified
-                    }));
-                    searchResult.add(id);
+                if (this.preferences.get('extensions.onlyShowVerifiedExtensions')) {
+                    const verified = await this.fetchVerifiedStatus(id, client, allVersions);
+                    this.updateExtensions(data, id, allVersions, !!verified);
+                } else {
+                    this.updateExtensions(data, id, allVersions);
+                    const fetchPromise = this.fetchVerifiedStatus(id, client, allVersions).then(verified => {
+                        let extension = this.getExtension(id);
+                        extension = this.setExtension(id);
+                        extension.update(Object.assign({
+                            verified: verified
+                        }));
+                    });
+                    fetchPromisesArray.push(fetchPromise);
                 }
             }
-            this._searchResult = searchResult;
+            Promise.all(fetchPromisesArray).then(async () => this.doChange(async () => {
+                await this.initialized;
+            }));
         }, token);
     }
 
+    protected async fetchVerifiedStatus(id: string, client: OVSXClient, allVersions: VSXAllVersions): Promise<boolean | undefined> {
+        const res = await client.query({ extensionId: id, extensionVersion: allVersions.version, includeAllVersions: true });
+        let verified = res.extensions?.[0].verified;
+        if (!verified && res.extensions?.[0].publishedBy.loginName === 'open-vsx') {
+            verified = true;
+        }
+        return verified;
+    }
+
+    protected updateExtensions(data: VSXSearchEntry, id: string, allVersions: VSXAllVersions, verified?: boolean): void {
```

Comment:
Why is this method not called `addExtension`?

Reply:
Agreed, seems more appropriate.